### PR TITLE
(#98) Getting rid of the `BlockingSocketServer` at all

### PR DESCRIPTION
**Pull request by iakunin**
This is for https://github.com/yegor256/cactoos-http/issues/98

@victornoel @paulodamaso I've found a solution that could help up to get rid of the `BlockingSocketServer`: https://stackoverflow.com/a/904609/3456163

Let's see if it works in rultor-environment.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#101](https://github.com/yegor256/cactoos-http/issues/101) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#101](https://github.com/yegor256/cactoos-http/issues/101) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/1c7d0b34-0212-4160-81a9-9e344a2b88bc); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/http/HtTimedWireTest.java [hunk 1]
```diff
@@ -60,20 +58,19 @@ public void worksFine() throws Exception {
...
-                    server.address().getHostAddress(),
-                    server.port()
-                ),
-                timeout
-            ).send(new InputOf("unused"));
-        }
+        final int port = 80;
+        new HtTimedWire(
+            new HtWire(
+                "192.168.0.0",
```
Comment (victornoel) [inline src/test/java/org/cactoos/http/HtTimedWireTest.java]: @iakunin are we sure this one can never be routable? Why not use the example one in the SO answer (`10.255.255.1`)?

File: src/test/java/org/cactoos/http/HtTimedWireTest.java [hunk 2]
```diff
@@ -60,20 +58,19 @@ public void worksFine() throws Exception {
...
+                port
```
Comment (victornoel) [inline src/test/java/org/cactoos/http/HtTimedWireTest.java]: @iakunin I think the port can stay inlined

File: src/test/java/org/cactoos/http/HtTimedWireTest.java (hunk 1 above)
Comment (iakunin) [reply src/test/java/org/cactoos/http/HtTimedWireTest.java]: @victornoel you're right: using `10.255.255.1` is more obvious, cause this one is in SO answer.

Replaced.

**Comment by iakunin** [comment]
> @iakunin great idea, I'm glad we found a simpler solution. I'm just worried about making this more explicit. Do we see any idea how to better document why this work? Maybe let's just leave the SO url as a comment to explain why we use this to test timeout?

@victornoel and I'm happy to suggest such simpler solution. I agree with you about the comment. I left one about artificial timeout.

File: src/test/java/org/cactoos/http/HtTimedWireTest.java (hunk 2 above)
Comment (iakunin) [reply src/test/java/org/cactoos/http/HtTimedWireTest.java]: @victornoel no problem. Inlined.

File: src/test/java/org/cactoos/http/HtTimedWireTest.java [hunk 3]
```diff
@@ -39,11 +39,12 @@
  * @checkstyle JavadocMethodCheck (500 lines)
  * @checkstyle JavadocVariableCheck (500 lines)
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
```
Comment (victornoel) [inline src/test/java/org/cactoos/http/HtTimedWireTest.java]: @iakunin please keep the exception like this close to the place where they happen

File: src/test/java/org/cactoos/http/HtTimedWireTest.java [hunk 4]
```diff
@@ -58,17 +59,16 @@ public void worksFine() throws Exception {
...
     public void failsAfterTimeout() throws Exception {
-        // @checkstyle MagicNumberCheck (2 lines)
         final long timeout = 100;
-        final int port = 80;
         new HtTimedWire(
             new HtWire(
-                "192.168.0.0",
-                port
+                // non-routable IP address artificially creates a timeout error
+                // @see: https://stackoverflow.com/a/904609/3456163
```
Comment (victornoel) [inline src/test/java/org/cactoos/http/HtTimedWireTest.java]: @iakunin I'm not sure using a javadoc annotation is a proper use of them in a normal comment like here, I think it's best if you remove it

Comment (iakunin) [reply src/test/java/org/cactoos/http/HtTimedWireTest.java]: @victornoel you're right: that's not appropriate. Removed.

File: src/test/java/org/cactoos/http/HtTimedWireTest.java (hunk 3 above)
Comment (iakunin) [reply src/test/java/org/cactoos/http/HtTimedWireTest.java]: @victornoel ok, that's not a problem

**Comment by iakunin** [comment]
> @iakunin one two small changes needed

I've done it. That's not a problem :)

**Comment by victornoel** [comment]
@iakunin thanks!
@paulodamaso it's good to merge

File: src/test/java/org/cactoos/http/HtTimedWireTest.java [hunk 5]
```diff
@@ -60,20 +59,20 @@ public void worksFine() throws Exception {
...
-                    server.port()
-                ),
-                timeout
-            ).send(new InputOf("unused"));
-        }
+        new HtTimedWire(
+            new HtWire(
+                // non-routable IP address artificially creates a timeout error
+                // see: https://stackoverflow.com/a/904609/3456163
+                "10.255.255.1",
```
Comment (paulodamaso) [inline src/test/java/org/cactoos/http/HtTimedWireTest.java]: @iakunin Let's put this comment in the method javadoc, avoiding these comment inside the method

Comment (iakunin) [reply src/test/java/org/cactoos/http/HtTimedWireTest.java]: @paulodamaso done

**Comment by iakunin** [comment]
@paulodamaso @victornoel let's have a look one more time, please

**Comment by paulodamaso** [comment]
@iakunin @victornoel Great work on this one guys

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#101](https://github.com/yegor256/cactoos-http/issues/101) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by sereshqua** [comment]
@iakunin please make sure you start your comment with the name of the user they are addressed to, [see](https://github.com/yegor256/cactoos-http/pull/101#issuecomment-552226423)

@victornoel issue from yesterday repeats again :)

**Comment by iakunin** [comment]
@sereshqua you’re right. My bad.

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by iakunin** [comment]
> @iakunin @victornoel Great work on this one guys

@paulodamaso @victornoel it's pleasure to work with you, guys :)

**Pull request merged**